tests/MongoDB.EntityFrameworkCore.FunctionalTests/Query/WhereTests.cs

```diff
@@ -251,6 +251,39 @@ public void Where_string_list_any()
         Assert.All(results, p => Assert.NotEmpty(p.mainAtmosphere));
     }
 
+    [Fact(Skip = "Errors wiuth MongoDB < 5.0")]
+    public void Where_entity_equal_null()
+    {
+        var actual = _db.Planets.FirstOrDefault(p => p == null);
+        Assert.Null(actual);
+    }
+
+    [Fact]
+    public void Where_entity_not_equal_null()
+    {
+        var actual = _db.Planets.First(p => p != null);
+        Assert.NotNull(actual);
+    }
+
+    [Fact]
+    public void Where_entity_equal_throws()
+    {
+        var first = _db.Planets.First();
+        var ex = Assert.Throws<NotSupportedException>(() => _db.Planets.Where(p => p == first).ToList());
```

Comment:
Could you please elaborate why we throws here? It looks like legitimate way to use for me.

Reply:
There's some debate/discussion as to what this should do.

The EF team's preference is to rewrite this as a id based comparison for items with keys and a full object comparison for items that don't have identity.

Robert has some reservations about that and it would mean that the C# Driver and the EF Provider diverge on behavior here.

If an item has changed in-memory do you expect to be able to get the original one still? With the EF method we would but with the C# Driver we won't.

The original bug this PR is addressing was really just around null and so I want to get this unblocked and shipped and then we can move on to discussion and resolution of actual entity comparison next release when had more time to discuss.